aiida_quantumespresso/cli/data/structure.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+"""Command line utilities to create and inspect `StructureData` nodes."""
+from __future__ import absolute_import
+
+import click
+
+from aiida.cmdline.params import options
+from aiida.cmdline.utils import decorators, echo
+
+from . import cmd_data
+
+
+@cmd_data.group('structure')
+def cmd_structure():
+    """Commands to create and inspect `StructureData` nodes."""
+
+
+@cmd_structure.command('import')
+@click.argument('filename', type=click.File('r'))
+@options.DRY_RUN()
+@decorators.with_dbenv()
+def cmd_import(filename, dry_run):
+    """Import a `StructureData` from a Quantum ESPRESSO input file."""
+    from aiida_quantumespresso.tools.pwinputparser import PwInputFile
```

Comment:
My guess is that this parser already encompasses the old `get_structuredata_from_qetools` and `get_kpointsdata_from_qetools`? :)

Reply:
Yes, the basic parsing functionality is now part of `qe-tools` and is fully AiiDA agnostic. In a recent PR in `aiida-quantumespresso` I extend that basic parsing functionality into transforming them into nodes. This simple CLI then simply exposes that through the CLI.

Reply:
Very elegant.